src/libratbag-data.c

```diff
@@ -251,6 +251,9 @@ init_data_steelseries(struct ratbag *ratbag,
 		str = g_key_file_get_string(keyfile, group, "DpiList", NULL);
 		if (str)
 			data->steelseries.dpi_list = dpi_list_from_string(str);
+
+		free(str);
+		str = NULL;
```

Comment:
this shouldn't be necessary because we mark `str` as `_cleanup_(freep)`. The issue here is that we're re-using it so the value gets overwritten but re-using auto-cleanup pointers is bad form. A better approach would be to have a separate variable for the dpi string and the quirk string below, both of hem with `_cleanup_(freep)`

also, indentation looks really messed up here in this block unless that's the github rendering.

Reply:
Hmm, I see. I don't know much about them because I have never used them myself. Thanks for an advice!
I remade this like you explained here.

About indentation, seems fine to me? Both in the editor and on GitHub.

Reply:
weird, it looks fine in the snippet above but in the diff view it was off. Anyway, all good then, thanks for the change.